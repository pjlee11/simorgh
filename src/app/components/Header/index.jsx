import React from 'react';
import Brand from '../Brand';

const Header = () => (
  <header role="banner">
    <Brand indentedLogo={false} />
  </header>
);

export default Header;
