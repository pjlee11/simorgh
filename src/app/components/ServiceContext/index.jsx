import React from 'react';
import { func, node, string } from 'prop-types';
import services from '../../lib/config/services';

/* Create ServiceContext using the default service */
export const ServiceContext = React.createContext(services.default);

export const ServiceContextProvider = ({ children, service }) => (
  <ServiceContext.Provider value={services[service]}>
    {children}
  </ServiceContext.Provider>
);

export const ServiceContextConsumer = ({ children }) => (
  <ServiceContext.Consumer>{children}</ServiceContext.Consumer>
);

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
};

ServiceContextConsumer.propTypes = {
  children: func.isRequired,
};
