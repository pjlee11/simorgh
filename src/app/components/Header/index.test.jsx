import React from 'react';
import Header from './index';
import { ServiceContextProvider } from '../ServiceContext';
import { shouldMatchSnapshot } from '../../helpers/tests/testHelpers';

describe(`Header`, () => {
  shouldMatchSnapshot(
    'should render correctly',
    <ServiceContextProvider service="default">
      <Header />
    </ServiceContextProvider>,
  );
});
