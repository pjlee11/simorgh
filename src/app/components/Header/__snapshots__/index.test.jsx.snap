// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
.c6 {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  overflow: hidden;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  top: 0;
}

.c0 {
  display: grid;
  background-color: #B80000;
  height: 5rem;
  width: 100%;
}

.c3 {
  padding-top: 0.5rem;
  display: inline-block;
}

.c4 {
  display: block;
  padding-bottom: 1.5rem;
}

.c2:hover .c4,
.c2:focus .c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.25rem solid #FFFFFF;
}

.c5 {
  display: block;
  height: 1.5rem;
  width: 10.496875rem;
  margin-top: 1.25rem;
}

.c1 {
  grid-column: 1 / -1;
}

@media (max-width:37.4375em) {
  .c0 {
    grid-gap: 0.5rem;
  }
}

@media (min-width:37.5em) {
  .c0 {
    grid-gap: 1rem;
  }
}

@media (max-width:25em) {
  .c0 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25em) and (max-width:62.9375em) {
  .c0 {
    padding: 0 1rem;
  }
}

@media (max-width:62.9375em) {
  .c0 {
    grid-template-columns: repeat(6,1fr);
  }
}

@media (min-width:63em) and (max-width:80em) {
  .c0 {
    grid-template-columns: 1fr repeat(8,minmax(0,6.75rem)) 1fr;
  }
}

@media (min-width:80em) {
  .c0 {
    grid-template-columns: 1fr repeat(10,minmax(0,6.9rem)) 1fr;
  }
}

@media (min-width:62.9375em) {
  .c0 {
    padding: 0 1rem;
  }
}

<header
  role="banner"
>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <a
        className="c2 c3"
        href="https://www.bbc.co.uk/news"
      >
        <span
          className="c4"
        >
          <svg
            aria-hidden="true"
            className="c5"
            focusable="false"
            viewBox="0 0 167.95 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="#fff"
              fillRule="evenodd"
            >
              <path
                d="M84.32,0V24H58.82V0ZM79.26,16l-.41.25a11.91,11.91,0,0,1-6.07,1.85c-4.18,0-6.93-2.49-6.94-6.09s2.88-6.13,6.83-6.14a12,12,0,0,1,6,1.71l.4.22V4.6l-.17-.07A16.54,16.54,0,0,0,72.7,3.14a10.56,10.56,0,0,0-7.19,2.57,8.68,8.68,0,0,0-2.85,6.54A8.46,8.46,0,0,0,65,17.93a10,10,0,0,0,7.5,2.94h0a14,14,0,0,0,6.56-1.5l.15-.07Z M54.94,0V24H29.43V0ZM48.88,15.91a4.62,4.62,0,0,0-3.77-4.47,4.25,4.25,0,0,0,1.49-1.19,3.75,3.75,0,0,0,.72-2.34,4.17,4.17,0,0,0-1.39-3.13,6.33,6.33,0,0,0-4.36-1.41H36.12V20.65H42.5a7.18,7.18,0,0,0,4.9-1.53A4.27,4.27,0,0,0,48.88,15.91Z M45.7,15.57A2,2,0,0,1,45,17.16a4.22,4.22,0,0,1-2.85.79H39.22V13.23H42a4.89,4.89,0,0,1,2.82.68A1.93,1.93,0,0,1,45.7,15.57Z M43.32,9.92a2,2,0,0,0,.82-1.79,1.82,1.82,0,0,0-.58-1.43,3.33,3.33,0,0,0-2.26-.63H39.22v4.47h1.43A4.62,4.62,0,0,0,43.32,9.92Z M25.55,0V24H0V0ZM19.49,15.91a4.62,4.62,0,0,0-3.77-4.47,4.24,4.24,0,0,0,1.49-1.19,3.75,3.75,0,0,0,.72-2.34,4.18,4.18,0,0,0-1.39-3.13,6.33,6.33,0,0,0-4.36-1.41H6.74V20.65h6.38A7.18,7.18,0,0,0,18,19.12,4.27,4.27,0,0,0,19.49,15.91Z M13.93,9.92a2,2,0,0,0,.82-1.79,1.82,1.82,0,0,0-.58-1.43,3.33,3.33,0,0,0-2.26-.63H9.83v4.47h1.43A4.62,4.62,0,0,0,13.93,9.92Z M15.47,13.91a4.89,4.89,0,0,0-2.82-.68H9.83V18h2.94a4.23,4.23,0,0,0,2.85-.79,2,2,0,0,0,.69-1.59A1.93,1.93,0,0,0,15.47,13.91Z"
              />
              <path
                d="M111,3.36h2.36V20.7h-2.13L99.66,7.35V20.7H97.33V3.36h2L111,16.83Z"
              />
              <path
                d="m117.86 3.36h9.83v2.21h-7.35v5.29h7.1v2.22h-7.1v5.39h7.58v2.21h-10.06z M154.18,3.36h2.48l-7,17.41h-.55l-5.67-14.1-5.73,14.1h-.53l-7-17.41h2.5l4.78,12,4.81-12h2.35l4.83,12Z M163.38,13.43l-1.89-1.15A8.57,8.57,0,0,1,159,10.16a4,4,0,0,1-.75-2.41,4.26,4.26,0,0,1,1.42-3.33,5.31,5.31,0,0,1,3.69-1.28,7,7,0,0,1,4,1.22V7.17a5.74,5.74,0,0,0-4-1.8,3.34,3.34,0,0,0-2,.56,1.71,1.71,0,0,0-.78,1.44,2.22,2.22,0,0,0,.58,1.46,7.25,7.25,0,0,0,1.85,1.43l1.9,1.12Q168,13.28,168,16.21a4.42,4.42,0,0,1-1.4,3.39A5.11,5.11,0,0,1,163,20.9a7.63,7.63,0,0,1-4.68-1.58V16.17a5.84,5.84,0,0,0,4.65,2.55,2.93,2.93,0,0,0,1.94-.65,2,2,0,0,0,.78-1.63Q165.67,14.85,163.38,13.43Z"
              />
            </g>
          </svg>
          <span
            className="c6"
          >
            Default Brand Name
          </span>
        </span>
      </a>
    </div>
  </div>
</header>
`;
